x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -106,29 +104,24 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
         }
     }
 
+    private void authenticateOnlyNioHttpRequests(HttpRequest request, ActionListener<Void> listener) {
+        if (request instanceof HttpPipelinedRequest) {
+            request = ((HttpPipelinedRequest) request).getDelegateRequest();
+        }
+        if (request instanceof NioHttpRequest) {
+            this.authenticationService.authenticate(
+                request,
+                ActionListener.wrap(ignored -> listener.onResponse(null), listener::onFailure)
+            );
+        } else {
+            listener.onResponse(null);
```

Comment:
Had to implement some convoluted get-arounds to address this, because besides `Netty4HttpRequest`s, in tests, there are also `FakeHttpRequests` inbound.